src/codegen/runtime_functions.cpp

```diff
@@ -129,5 +137,20 @@ void RuntimeFunctions::ThrowOverflowException() {
   throw std::overflow_error("ERROR: overflow");
 }
 
+util::IndexScanIterator *RuntimeFunctions::GetIterator(index::Index *index,
```

Comment:
Can we have some helper for the index scan translator, such as `sorter.cpp` and `inserter.cpp`?  Having this and `DeleteIterator()` without some _holder_ object would be dangerous and makes the code less readable.  It may cause memory leaks, if there is any exception thrown between them, and it is hard to catch them later.

i.e. make a helper object and move these into the helper.

Reply:
Hey Tony, the iterator is already a helper object; I guess what you mean is to construct the iterator with `context.GetRuntimeState().RegisterState()` function in the index_scan_translator constructor instead of calling the `GetIterator()` and `DeleteIterator()` function, right?

Reply:
@huangminchn Yes. The point is on the lifetime of this iterator object. The lifetime of the iterator should be the same or shorter than the one registered with the runtime state, after all.